src/test/java/org/zeromq/ZMsgTest.java

```diff
@@ -123,20 +122,18 @@ public void testHashcode()
 
         assertThat(msg.hashCode(), is(other.hashCode()));
 
-        other = new ZMsg();
-        other.add("2");
+        other = new ZMsg().add("2");
 
         assertThat(msg.hashCode(), is(not(equalTo(other.hashCode()))));
     }
 
     @Test
     public void testDump()
     {
-        ZMsg msg = new ZMsg();
-
-        msg.add(new byte[0]);
-        msg.add(new byte[] { (byte) 0xAA });
-        msg.dump();
+        ZMsg msg = new ZMsg()
+                .add(new byte[0])
+                .add(new byte[] { (byte) 0xAA })
+                .dump();
```

Comment:
Generally I'm in favor of the builder pattern, but it feels a little weird to me to include `.dump()` in the chain here, since that is an action that you perform using the ZMsg you've built. I would write it like this:

```java
ZMsg msg = new ZMsg()
        .add(new byte[0])
        .add(new byte[] { (byte) 0xAA });

msg.dump();

StringBuilder out = new StringBuilder();
msg.dump(out);
```

Reply:
I think dump method usually used for log or debug purpose, I consider it something like peek() method in java8 streams, where you want to do some other things in the middle of work without any side-effect. Maybe I'm wrong, however I don't insist on builder pattern for dump method.

Reply:
I think it's probably valuable to have the builder pattern available for `dump` (it can't hurt, right?), but perhaps we could change the usage example here in the tests to reflect recommended usage.

Reply:
I Agree